### ISSUE TITLE
Amazon S3 upload not working

### ISSUE THREAD

**Issue by andrewhavens** [refinery/refinerycms#3023]
I'm trying to use Amazon S3 for uploads, but it isn't working. I have set the environment variables, and also set them in the initializer, but it still isn't working. I just see blank images:

<img width="485" alt="screen shot 2015-08-01 at 7 28 55 pm" src="https://cloud.githubusercontent.com/assets/446392/9024430/c9ec3206-3883-11e5-92ea-9578bc49e3ba.png">

Anybody else experiencing this? Or know of a setting that I'm missing?

FYI: I'm using the master branch of Refinery, but it not the latest commit.

**Comment by andrewhavens** [comment]
Here is the log output:

```
I, [2015-08-01T22:25:50.963440 #13983]  INFO -- : Started POST "/admin/images?dialog=true" for 73.164.205.208 at 2015-08-01 22:25:50 -0400
I, [2015-08-01T22:25:50.966225 #13983]  INFO -- : Processing by Refinery::Admin::ImagesController#create as HTML
I, [2015-08-01T22:25:50.966620 #13983]  INFO -- :   Parameters: {"utf8"=>"✓", "authenticity_token"=>"/gHuuok8xNJDtRsN83XKL7x/3LKh7z0GXYJP+Sk2iFBSSCSwdJkusSQl0XOdYYt1jJZ7CrZz+bggyrZLPR5z3A==", "image"=>{"image"=>[#<ActionDispatch::Http::UploadedFile:0x007fbd55fd4cf0 @tempfile=#<Tempfile:/tmp/RackMultipart20150801-13983-1hghqzk.jpg>, @original_filename="family.jpg", @content_type="image/jpeg", @headers="Content-Disposition: form-data; name=\"image[image][]\"; filename=\"family.jpg\"\r\nContent-Type: image/jpeg\r\n">], "image_title"=>"", "image_alt"=>""}, "wymeditor"=>"", "modal"=>"true", "dialog"=>"true", "app_dialog"=>"true", "field"=>"", "update_image"=>"", "thumbnail"=>"", "callback"=>"", "conditions"=>"", "locale"=>:en}
D, [2015-08-01T22:25:50.969480 #13983] DEBUG -- :   Refinery::Authentication::Devise::User Load (0.6ms)  SELECT  "refinery_authentication_devise_users".* FROM "refinery_authentication_devise_users" WHERE "refinery_authentication_devise_users"."id" = $1  ORDER BY "refinery_authentication_devise_users"."id" ASC LIMIT 1  [["id", 1]]
D, [2015-08-01T22:25:50.972618 #13983] DEBUG -- :   Refinery::Authentication::Devise::Role Load (0.5ms)  SELECT "refinery_authentication_devise_roles".* FROM "refinery_authentication_devise_roles" INNER JOIN "refinery_authentication_devise_roles_users" ON "refinery_authentication_devise_roles"."id" = "refinery_authentication_devise_roles_users"."role_id" WHERE "refinery_authentication_devise_roles_users"."user_id" = $1  [["user_id", 1]]
D, [2015-08-01T22:25:50.980456 #13983] DEBUG -- :    (0.2ms)  BEGIN
D, [2015-08-01T22:25:51.749478 #13983] DEBUG -- :   SQL (0.8ms)  INSERT INTO "refinery_images" ("image_name", "image_size", "image_uid", "image_title", "image_alt", "created_at", "updated_at") VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING "id"  [["image_name", "family.jpg"], ["image_size", 21056], ["image_uid", "2015/08/01/22/25/51/204/family.jpg"], ["image_title", "Family"], ["image_alt", ""], ["created_at", "2015-08-02 02:25:51.743262"], ["updated_at", "2015-08-02 02:25:51.743262"]]
D, [2015-08-01T22:25:51.753443 #13983] DEBUG -- :   Refinery::Image Load (0.7ms)  SELECT  "refinery_images".* FROM "refinery_images"  ORDER BY "refinery_images"."id" ASC LIMIT 500
D, [2015-08-01T22:25:51.758671 #13983] DEBUG -- :    (0.8ms)  COMMIT
I, [2015-08-01T22:25:51.762141 #13983]  INFO -- :   Rendered refinery/admin/dialog_success.html.erb within layouts/refinery/admin_dialog (0.6ms)
I, [2015-08-01T22:25:51.762718 #13983]  INFO -- :   Rendered refinery/_html_tag.html.erb (0.1ms)
I, [2015-08-01T22:25:51.764137 #13983]  INFO -- :   Rendered refinery/admin/_dialog_success.html.erb (0.4ms)
I, [2015-08-01T22:25:51.764684 #13983]  INFO -- : Completed 200 OK in 798ms (Views: 4.6ms | ActiveRecord: 4.9ms)
```

**Comment by andrewhavens** [comment]
I logged into AWS and can confirm that the file was uploaded, but it is not being displayed in the admin.

**Comment by andrewhavens** [comment]
I also seem to be having the same problem when I disable using S3. Normal image uploads are not working for me. Could be a problem with my nginx configuration, but since S3 uploads have the same behavior, I don't think that nginx is to blame.

**Comment by parndt** [comment]
Hmm this is working for me without S3 in development mode on the latest commit - is it also failing for you with that? Is imagemagick working correctly?

**Comment by andrewhavens** [comment]
Alright I think I have it all figured out...you were right about ImageMagick. I thought I had already installed it, but I had not. Once I installed ImageMagick, the images started appearing. Strange that there were no error messages.

For Amazon S3 uploads...the only way that I can tell that it is working is by inspecting the bucket after uploading the file in the admin. It looks like it's working but there are no logs, and the image URL looks the same, even though it's being served from Amazon S3...does that sound correct?.
